src/yukon/backend/src/api/controllers/nodes/__init__.py

```diff
@@ -1,43 +1,136 @@
-from quart import Blueprint, jsonify, request
-from typing import Any, Dict
+from quart import Blueprint, jsonify, request, Response
+from typing import Any, Dict, Tuple
 
 nodes_controller = Blueprint('nodes', __name__)
 
+import asyncio
+import pathlib
+import tempfile
+
+import os
+import sys
+import pyuavcan
+import pyuavcan.transport.udp
+
+dsdl_generated_dir = pathlib.Path(tempfile.gettempdir(), 'dsdl', f'pyuavcan-v{pyuavcan.__version__}')
+dsdl_generated_dir.mkdir(parents=True, exist_ok=True)
+sys.path.insert(0, str(dsdl_generated_dir))
+
+import pyuavcan.application
+import uavcan.node                      # noqa E402
+
+
+class BusMonitorUtil():
+    def __init__(self, node_list=list(), nodeid_list=list()):
+        # Array of online nodes and its IDs
+        self._node_list = node_list
+        self._nodeid_list = nodeid_list
+
+    @property
+    def node_list(self):
```

Comment:
this returns a mutable reference which breaks encapsulation.

Reply:
Yeah this is a bit tricky here as lists are mutable structures. Do you have another suggestion?

Reply:
make it a method called `get_node_list_ref` to be explicit that a mutable reference is returned from the method. One expects properties to provide encapsulation. Don't just use them for syntactic sugar.